src/components/header/header.jsx

```diff
@@ -141,26 +142,41 @@ class Header extends Component {
 
   connectionConnected = () => {
     this.setState({ account: store.getStore('account') })
+    this.setAddressEnsName();
   };
 
   connectionDisconnected = () => {
     this.setState({ account: store.getStore('account') })
   }
 
+  setAddressEnsName = async () => {
+    const provider = store.getStore('web3context').library.provider
+    const account = store.getStore('account')
+    const { address } = account
+    const network = provider.networkVersion
+    const ens = new ENS({ provider, network })
+    const addressEnsName = await ens.reverse(address).catch(() => {})
+    if (addressEnsName) {
+      this.setState({ addressEnsName })
```

Comment:
Does not affect store state in any way. The intention is to minimize risk of regression as much as possible. Merging ENS info into account state is explicitly avoided.